src/components/MetaData/index.tsx

```diff
@@ -439,18 +486,17 @@ const MetaData: React.FC<Props> = ({
     </>
   );
 
-  // Canonical Link
-  const canonicalDomainMapping = {
-    article: canonicalDomain,
-    video: canonicalDomain,
-    gallery: canonicalDomain,
-  };
+  if (outputCanonicalLink) {
+    const defaultResolution = getPageCanonicalUrl(pageType, canonicalDomain || websiteDomain, gc);
```

Comment:
`canonicalDomain || websiteDomain` this means the calling code will need to do logic based on page type too. Otherwise the domain address that is used will be incorrect for different page types. Shouldn't the function itself handle which is used based on page type?

Reply:
We can't know what the client will want to override the `websiteDomain` with.  If they do want a unique domain for specific `pageTypes`, they can override it it in the `canonicalResolver` function.  Unless I'm not understanding where you are coming from.